lib/src/configuration.dart

```diff
@@ -27,14 +28,21 @@ class Configuration {
   final Map<String, ConfiguredValue> _values;
 
   /// Creates an implicit configuration with the given [values].
-  Configuration.implicit(this._values);
+  Configuration.implicit(this._values, {String? opaqueId})
```

Comment:
Rather than passing in the opaque ID itself, it might be better to make the field private, pass in `originalConfiguration`, and expose it publicly as a `bool sameOriginal(Configuration other)`. That way you encapsulate the details of exactly how we track originality, and expose only the core logic the caller needs to specify "this is the same as that".